### PR TITLE
Add precise keyword to the depth and forward pass shaders. 

### PR DESCRIPTION
**Pull request by rgba16f**
Modify metal shader pipeline to pass the invariant along to msl.

Signed-off-by: rgba16f <82187279+rgba16f@users.noreply.github.com>

### CODE REVIEW

**Review by invertednormal — APPROVED** [review]

File: Gems/Atom/RHI/Metal/Code/Source/RHI.Builders/ShaderPlatformInterface.cpp [hunk 1]
```diff
@@ -424,7 +424,7 @@ namespace AZ
             AZStd::string outMetalLibFile = RHI::BuildFileNameWithExtension(shaderSourceFile, tempFolder, "metallib");
             
             //Debug symbols are always enabled at the moment. Need to turn them off for optimized shader assets. 
-            AZStd::string shaderDebugInfo = "-gline-tables-only -MO";
+            AZStd::string shaderDebugInfo = "-gline-tables-only -MO -fpreserve-invariance";
```
Comment (VickyAtAZ) [inline Gems/Atom/RHI/Metal/Code/Source/RHI.Builders/ShaderPlatformInterface.cpp:427]: Will the -fpreserve-invariance be turned off when debug symbols is turned off for optimization in the future?

Comment (rgba16f) [reply Gems/Atom/RHI/Metal/Code/Source/RHI.Builders/ShaderPlatformInterface.cpp:427]: Probably not

Comment (rgba16f) [reply Gems/Atom/RHI/Metal/Code/Source/RHI.Builders/ShaderPlatformInterface.cpp]: Updated the PR with the -fpreserve-invariance moved to a separate msl to air options string.

File: Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/Surfaces/BasePbrSurfaceData.azsli [hunk 2]
```diff
@@ -24,7 +24,7 @@ static const float MinRoughnessA = 0.0005f;
 
 class BasePbrSurfaceData
 {
-    float3 position;            //!< Position in world-space
+    precise float3 position;            //!< Position in world-space
```
Comment (VickyAtAZ) [inline Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/Surfaces/BasePbrSurfaceData.azsli:27]: nit: comment alignment.

File: Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/Surfaces/StandardSurface.azsli [hunk 3]
```diff
@@ -20,7 +20,7 @@ class Surface
 
     // ------- BasePbrSurfaceData -------
     
-    float3 position;            //!< Position in world-space
+    precise float3 position;            //!< Position in world-space
```
Comment (VickyAtAZ) [inline Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/Surfaces/StandardSurface.azsli:23]: nit: comment alignment

**Review by VickyAtAZ — APPROVED** [review]

**Pull request merged**